src/components/image/Image.base.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { PImage } from './Image.props';
+
+const ImageBase: React.FC<PImage> = ({ source, width, height, alt, className }: PImage) => {
+    if (!source) {
+        throw new Error('Compass Components: You need to provide image source');
+    }
+
+    const [loading, setLoading] = useState(true);
+    const [image, setImage] = useState('');
+
+    const fetchUrl = useCallback(async () => {
+        const response = await fetch(source);
+        const imgUrl = await response.url;
+
+        setLoading(false);
+        setImage(imgUrl);
+    }, [source]);
+
+    useEffect(() => {
+        fetchUrl();
+    }, [source, fetchUrl]);
```

Comment:
if the function is defined within the `useEffect` hook you do not need to set a dependency here and you can also remove the `useCallback` wrapping the fetch function